commit b245a6ee83a16b3499225dca5936242f423ffe66
Author: Marco Martin <notmart@gmail.com>
Date:   Tue Aug 19 17:09:57 2014 +0200

    Fix installing new .comic packages from GHNS
    
    When installing a new .comic provider from GHNS, it doesn't appear in
    the installed list on the comic widget.
    
    patch by Andrei Amuraritei

diff --git a/plasma/tools/plasmapkg/main.cpp b/plasma/tools/plasmapkg/main.cpp
index 61492fe..fdbeb0b 100644
--- a/plasma/tools/plasmapkg/main.cpp
+++ b/plasma/tools/plasmapkg/main.cpp
@@ -38,6 +38,7 @@
 #include <Plasma/PackageStructure>
 #include <Plasma/Package>
 #include <Plasma/PackageMetadata>
+#include <Plasma/Theme>
 
 static const char description[] = I18N_NOOP("Install, list, remove Plasma packages");
 static const char version[] = "0.2";
@@ -401,18 +402,14 @@ int main(int argc, char **argv)
             }
 
             QStringList installed = packages(pluginTypes);
-            if (installed.contains(pluginName)) {
-                if (installer->uninstallPackage(pluginName, packageRoot)) {
-                    output(i18n("Successfully removed %1", pluginName));
-                } else if (!args->isSet("upgrade")) {
-                    output(i18n("Removal of %1 failed.", pluginName));
-                    delete installer;
-                    return 1;
+            //Get list of installed desktop themes with Plasma::Theme::listThemeInfo
+            if (type.compare(i18nc("package type", "theme"), Qt::CaseInsensitive) == 0 ||
+                type.compare("theme", Qt::CaseInsensitive) == 0) {
+                foreach (KPluginInfo theme, Plasma::Theme::listThemeInfo()) {
+                      installed << theme.pluginName();
                 }
-            // Need to check for theme package type, and uninstall anyway, since they aren't returned
-            // in the list of plugins. Fixes https://bugs.kde.org/show_bug.cgi?id=149479
-            } else if (type.compare(i18nc("package type", "theme"), Qt::CaseInsensitive) == 0 ||
-                       type.compare("theme", Qt::CaseInsensitive) == 0) {
+            }
+            if (installed.contains(pluginName)) {
                 if (installer->uninstallPackage(pluginName, packageRoot)) {
                     output(i18n("Successfully removed %1", pluginName));
                 } else if (!args->isSet("upgrade")) {
@@ -420,9 +417,9 @@ int main(int argc, char **argv)
                     delete installer;
                     return 1;
                 }
-            }else {
-                output(i18n("Plugin %1 is not installed.", pluginName));
-                return 1;
+            }
+            else {
+                output(i18n("Plugin %1 is not installed now.", pluginName));
             }
         }
         if (args->isSet("install") || args->isSet("upgrade")) {
